nullaway/src/test/java/com/uber/nullaway/EnsuresNonNullIfTests.java

```diff
@@ -0,0 +1,898 @@
+            "    if(nullableItem == null || nullableItem2 == null) {",
+            "        return true;",
+            "    } else {",
+            "        return false;",
+            "    }",
+            "  }",
+            "  public int runOk() {",
+            "    if(nullableItemsAreNull()) {",
+            "      return 1;",
+            "    }",
+            "    nullableItem.call();",
+            "    nullableItem2.call();",
+            "    return 0;",
+            "  }",
+            "}")
+        .addSourceLines(
+            "Item.java", "package com.uber;", "class Item {", "  public void call() { }", "}")
+        .doTest();
+  }
+
+  @Test
+  public void setResultToFalse_multipleElements_correctSemantics_dereferenceFound() {
+    defaultCompilationHelper
+        .addSourceLines(
+            "Foo.java",
+            "package com.uber;",
+            "import javax.annotation.Nullable;",
+            "import com.uber.nullaway.annotations.EnsuresNonNullIf;",
+            "class Foo {",
+            "  @Nullable Item nullableItem;",
+            "  @Nullable Item nullableItem2;",
+            "  @EnsuresNonNullIf(value={\"nullableItem\",\"nullableItem2\"}, result=false)",
+            "  public boolean nullableItemsAreNull() {",
+            "    // If the function returns false, we know that neither of the fields can be null, i.e., both are non-null.",
+            "    return nullableItem == null || nullableItem2 == null;",
+            "  }",
+            "  public int runOk() {",
+            "    // BUG: Diagnostic contains: dereferenced expression nullableItem is @Nullable",
+            "    nullableItem.call();",
+            "    // BUG: Diagnostic contains: dereferenced expression nullableItem2 is @Nullable",
+            "    nullableItem2.call();",
+            "    return 0;",
+            "  }",
+            "}")
+        .addSourceLines(
+            "Item.java", "package com.uber;", "class Item {", "  public void call() { }", "}")
+        .doTest();
+  }
+
+  /********************************************************
+   * Tests related to semantic issues
+   ********************************************************
+   */
+  @Test
+  public void semanticIssues_doesntEnsureNonNullabilityOfFields() {
+    defaultCompilationHelper
+        .addSourceLines(
+            "Foo.java",
+            "package com.uber;",
+            "import javax.annotation.Nullable;",
+            "import com.uber.nullaway.annotations.EnsuresNonNullIf;",
+            "class Foo {",
+            "  @Nullable Item nullableItem;",
+            "  @EnsuresNonNullIf(value=\"nullableItem\", result=true)",
+            "  public boolean hasNullableItem() {",
+            "    if(nullableItem != null) {",
+            "      // BUG: Diagnostic contains: The method ensures the non-nullability of the fields, but returns incorrectly",
```

Comment:
Same comment as the one below.

Reply:
Here https://github.com/uber/NullAway/pull/1044/commits/56a0984a935cc89a0dd7819590afe6c7f4bd481e